platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -174,86 +181,61 @@ private void installFeatures(FeaturesService featuresService, List<String> insta
-                            featureObject.getName(), featureObject.getVersion(), NO_AUTO_REFRESH);
-                      } catch (Exception e) {
-                        printItemStatusFailure("Uninstall Failed: ", feature);
-                        throw e;
-                      }
-                      return true;
-                    });
-                printItemStatusSuccess("Uninstalled: ", feature);
-              });
+      Set<String> uniqueValues = new HashSet<>();
+      for (String feature : uninstallFeatures) {
+        if (uniqueValues.add(feature)) {
+          printItemStatusPending("Uninstalling: ", feature);
+          Feature featureObject = null;
+          try {
+            featureObject = featuresService.getFeature(feature);
+          } catch (Exception e) {
+            printError(String.format("Feature: %s does not exist", feature));
+            throw e;
+          }
+          if (featureObject == null) {
+            printItemStatusFailure("Uninstall Failed: ", feature);
+            printError(String.format("Feature: %s does not exist", feature));
+            throw new IllegalArgumentException(
+                String.format("Feature: %s does not exist", feature));
+          }
+          uninstallFeature(featuresService, featureObject);
+          printItemStatusSuccess("Uninstalled: ", feature);
+        }
+      }
     }
   }
 
   private void stopBundles(BundleService bundleService, List<String> stopBundleNames)
       throws BundleException, IllegalArgumentException {
     if (stopBundleNames != null) {
       printSectionHeading("Stopping Bundles");
-      stopBundleNames
-          .stream()
-          .distinct()
-          .forEach(
-              bundle -> {
-                printItemStatusPending("Stopping: ", bundle);
-                executeAsSystem(
-                    () -> {
-                      try {
-                        bundleService.getBundle(bundle).stop();
-                      } catch (BundleException e) {
-                        printItemStatusFailure("Stop Failed: ", bundle);
-                        throw e;
-                      } catch (IllegalArgumentException ie) {
-                        printItemStatusFailure("Stop Failed: ", bundle);
-                        printError(String.format("Bundle: %s does not exist", bundle));
-                        throw ie;
-                      }
-                      return true;
-                    });
-                printItemStatusSuccess("Stopped: ", bundle);
-              });
+      Set<String> uniqueValues = new HashSet<>();
+      for (String bundle : stopBundleNames) {
+        if (uniqueValues.add(bundle)) {
+          printItemStatusPending("Stopping: ", bundle);
+          try {
+            stopBundle(bundleService, bundle);
+          } catch (BundleException e) {
+            throw e;
+          }
+          printItemStatusSuccess("Stopped: ", bundle);
+        }
+      }
     }
   }
 
```

Comment:
❗️ We should audit the reading of the profile file from disk below.

Reply:
Done